### PR TITLE
Remove e2e/go.mod

### PR DESCRIPTION
**Pull request by nunnatsa**
e2e/go.mod was added to split the e2e into a new module, to allow using older k8s packages in e2e tests, because the new k8s packages conflicted with the KubeVirt go-client.

This PR replaces the KV go-client with a the kubernetes client, and so it is now possible to use the newest k8s packages, and in general, the same packages as the application uses. 

Also made some smaller fixes and improvements in the e2e tests, related to contexes and assertions.

**Release notes**:

```release-note
none
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -198,6 +204,16 @@ generate-manifests: $(CONTROLLER_GEN) ## Generate manifests e.g. CRD, RBAC etc.
...
+.PHONY: generate-kv-go-client
+generate-kv-go-client: $(CLIENT_GEN) ## Generate kubevirt client-go for e2e
+	$(CLIENT_GEN) \
+		--input-base="kubevirt.io/api" \
+		--input="core/v1" \
+		--output-pkg="sigs.k8s.io/cluster-api-provider-kubevirt/e2e/generated/client-go/clientset" \
+		--clientset-name="versioned" \
+		--output-dir /home/nunnatsa/GIT/cluster-api-provider-kubevirt/e2e/generated/client-go/clientset \
+		--fake-clientset=false
+
```
Comment (davidvossel) [inline Makefile:216]: do we need the kubevirt client once you transition e2e to use virtctl?

Comment (nunnatsa) [reply Makefile:216]: I'll remove kv client. this is still WIP.

File: Makefile [hunk 2]
```diff
@@ -225,7 +225,7 @@ docker-build: docker-pull-prerequisites ## Build the docker image for controller
 
 .PHONY: docker-push
 docker-push: ## Push the docker image
-	docker push $(CONTROLLER_IMG)-$(ARCH):$(TAG)
+	docker push --tls-verify=false $(CONTROLLER_IMG)-$(ARCH):$(TAG)
```
[the remark this answers is not in the document]
Comment (awels) [reply Makefile:228]: I am assuming you added this for when pushing to a repo that doesn't have a proper signed cert. What I did in HPP is make it an argument something like this:
```
TLS_VERIFY?=true
...

docker push -tls-verify=${TLS_VERIFY} $(CONTROLLER_IMG)-$(ARCH):$(TAG)
```

This way if you are testing locally you can just `export TLS_VERIFY=false` and it works. But it defaults to true.

**Pull request merged**